### PR TITLE
added registration/all users/delete user/add product/all products/add…

### PR DESCRIPTION
**Pull request by TimoxaV**
… to cart/products in cart

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.math.BigDecimal;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/AddProductController.java:16]: Let's make all of the service objects not final

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String priceString = req.getParameter("price");
+        BigDecimal price = BigDecimal.valueOf(Long.parseLong(priceString));
+        Product newProduct = new Product(name, price);
+        productService.create(newProduct);
+        resp.sendRedirect(req.getContextPath() + "/");
+        //req.getRequestDispatcher("/WEB-INF/views/product/addProduct.jsp").forward(req, resp);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/AddProductController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/product/allProduct.jsp").forward(req, resp);
```

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/product/addProduct.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/AddProductController.java:27]: What happens if we receive the empty Strings?
Will you be able to work with them?

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.valueOf(userId);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/DeleteUserController.java:21]:
```suggestion
        Long id = Long.valueOf(req.getParameter("id"));
```

File: src/main/java/com/internet/shop/controllers/RegistrationController.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/registration.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/RegistrationController.java:29]: Same check of input data

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Couldn't find the method that deletes a product from the user's cart

File: src/main/java/com/internet/shop/controllers/AddProductController.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/AddProductController.java:16]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/homework/lesson_16_hw.html
In this example service is final. So, I guess it should be final :)

File: src/main/java/com/internet/shop/controllers/AddProductController.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/AddProductController.java:27]: Checklist says: "All checks to see if any field is empty or that field should accept only numbers should be done by jsp"
All fields on the jsp must be "required".

File: src/main/java/com/internet/shop/controllers/AddProductController.java (hunk 2 above)
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/AddProductController.java]: do we need commented code here?

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        userService.delete(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/DeleteUserController.java:22]: let's delete shoppingCart here as well

File: src/main/java/com/internet/shop/controllers/ProductController.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ProductController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShoppingCartController.java]:
```suggestion
public class GetShoppingCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/UserController.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/UserController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/injectData.jsp [hunk 10]
```diff
@@ -0,0 +1,18 @@
+<%--
+  Created by IntelliJ IDEA.
+  User: Vadim
```
Comment (RomanDubovskyi) [inline src/main/webapp/WEB-INF/views/injectData.jsp]: let's remove autogenerated by idea comments

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -10,6 +10,86 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addToCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductToCart</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addToCart</servlet-name>
+        <url-pattern>/cart/add</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cart/products/add</url-pattern>
```

**Review by RomanDubovskyi — COMMENTED** [review]
let's add possibility to remove product from shoppingCart as well

File: src/main/java/com/internet/shop/controllers/AddProductController.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/AddProductController.java]: removed

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java (hunk 6 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/DeleteUserController.java:22]: done

File: src/main/java/com/internet/shop/controllers/ProductController.java (hunk 7 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/ProductController.java]: done

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java (hunk 8 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: done

**Comment by RomanDubovskyi** [comment]
my bad, that's task from next hw, will except task without it 
image

File: src/main/java/com/internet/shop/controllers/UserController.java (hunk 9 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/UserController.java]: done

File: src/main/webapp/WEB-INF/views/injectData.jsp (hunk 10 above)
Comment (TimoxaV) [reply src/main/webapp/WEB-INF/views/injectData.jsp]: done

File: src/main/webapp/web.xml (hunk 11 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/shop/controllers/AddProductToCart.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductToCart extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/AddProductToCart.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

**Review by RomanDubovskyi — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/AddProductToCart.java (hunk 12 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/AddProductToCart.java]: done

**Pull request merged**